contracts/FungibleTokenMetadataViews.cdc

```diff
@@ -0,0 +1,177 @@
+/// 
+/// Structs and resources can implement one or more
+/// metadata types, called views. Each view type represents
+/// a different kind of metadata.
+///
+pub contract FungibleTokenMetadataViews {
+    /// FTView wraps FTDisplay and FTVaultData, and is used to give a complete 
+    /// picture of a Fungible Token. Most Fungible Token contracts should 
+    /// implement this view.
+    ///
+    pub struct FTView {
+        pub let ftDisplay: FTDisplay?     
+        pub let vaultData: FTVaultData?
+        init(
+            ftDisplay: FTDisplay?,
+            vaultData: FTVaultData?
+        ) {
+            self.ftDisplay = ftDisplay
+            self.vaultData = vaultData
+        }
+    }
+
+    /// Helper to get a FT view 
+    ///
+    /// @param viewResolver: A reference to the resolver resource
+    /// @return A FTView struct
+    ///
+    pub fun getFTView(viewResolver: &{MetadataViews.Resolver}): FTView {
+        let ftView = viewResolver.resolveView(Type<FTView>())
+        if ftView != nil {
+            return ftView! as! FTView
+        }
+        return FTView(
+            ftDisplay: self.getFTDisplay(viewResolver),
+            vaultData : self.getFTVaultData(viewResolver)
+        )
+    }
+
+    /// View to expose the information needed to showcase this FT. 
+    /// This can be used by applications to give an overview and 
+    /// graphics of the FT.
+    ///
+    pub struct FTDisplay {
+        /// Name that should be used when displaying this FT.
+        pub let name: String
+
+        /// Description that should be used to give an overview of this FT.
+        pub let description: String
+
+        /// A small thumbnail representation of the object.
+        ///
+        /// This field should be a web-friendly file (i.e JPEG, PNG)
+        /// that can be displayed in lists, link previews, etc.
+        pub let thumbnail: AnyStruct{MetadataViews.File}
+
+        /// External link to a URL to view more information about this collection.
+        pub let externalURL: MetadataViews.ExternalURL
+
+        /// Square-sized image to represent this collection.
+        pub let squareImage: MetadataViews.Media
+
+        /// Banner-sized image for this collection, recommended to have a size near 1200x630.
+        pub let bannerImage: MetadataViews.Media
+
+        /// Social links to reach this collection's social homepages.
+        /// Possible keys may be "instagram", "twitter", "discord", etc.
+        pub let socials: {String: MetadataViews.ExternalURL}
+
+        init(
+            name: String,
+            description: String,
+            thumbnail: AnyStruct{MetadataViews.File},
+            externalURL: MetadataViews.ExternalURL,
+            squareImage: MetadataViews.Media,
+            bannerImage: MetadataViews.Media,
+            socials: {String: MetadataViews.ExternalURL}
+        ) {
+            self.name = name
```

Comment:
What about the token symbol?

Reply:
I wanna discuss over the FLIP on the forum about the fields of this `FTDisplay`, the symbol is a really good catch, and I also would like to discuss which images we really do need for the FTs, I'm not sure at all that we need a square and a banner Image

Reply:
can you please mention the flip here